### PR TITLE
Issue #7853 - fix regression that allowed heading, pitch, roll input numbers to exceed given range

### PR DESCRIPTION
**Pull request by tonym3nsa**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7853 

**What is the new behavior?**

Heading, pitch and roll input do not allow numbers that exceed their given ranges
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/share/SharePanel.jsx [hunk 1]
```diff
@@ -152,6 +152,13 @@ class SharePanel extends React.Component {
             !isEqual(this.props?.viewerOptions?.orientation, newProps.viewerOptions?.orientation)) {
```
Comment (dsuren1) [inline web/client/components/share/SharePanel.jsx:152]: I think you need to revisit this condition and below code changes.

Maybe you can remove this condition and skip the state update of orientation fields from [here](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/share/SharePanel.jsx#L195-L197), as you are doing it explicitly below

**Review by dsuren1 — CHANGES_REQUESTED** [review]
Hi @tonym3nsa 
I noticed the issue mentioned by @tdipisa [here](https://github.com/geosolutions-it/MapStore2/pull/7913#issuecomment-1061702800) is happening again.

When changing the orientation of the map with 3d map tools
- [ ] The orientation value is updated incorrectly Ex: roll exceeding the range
- [ ] The orientation value is not in degrees but in radian

Please check this gif for the steps to reproduce
share_3d_radian

**Comment by tonym3nsa** [comment]
@tdipisa @dsuren1 So, at this point, I'm able to limit heading, pitch, and the roll value using SharePanel and once the map is rotated using Cesium navigation, I return either the minimum value or maximum value depending on which value is exceeded. However, the rotation of the map using the compass doesn't seem that it can be stopped by minimum and maximum - I've checked on Cesium and Cesium navigation's documentation with no success - that may require further investigation.

**Review by dsuren1 — CHANGES_REQUESTED** [review]
@tonym3nsa 
This is what I observed.
- [x] The value is limited when it exceeds the range, using 3d map compass, and shown in the share panel 
- [ ] The only issue now I see is everytime I use compass, the values are shown in **radian** in the share panel or sometimes **both** ex: _roll in **degress** and heading in **radian**_, which is confusing and it needs to be fixed.
2022-03-09 18_02_53-MapStore HomePage

@tdipisa 
I don't see any customization support for the **3d map compass tool** to limit it's rotations. So, an issue to investigate applying limits on the compass is not necessary, as we are showing only values within range in share panel. Let me know your thoughts.

**Comment by dsuren1** [comment]
@tonym3nsa 
A small advice, please make sure to add new commits to the PR whenever you are making changes and don't force push as it makes difficult to track the modifications done

**Comment by tonym3nsa** [comment]
> @tonym3nsa This is what I observed.

> * [ ]  The only issue now I see is everytime I use compass, the values are shown in **radian** in the share panel or sometimes **both** ex: _roll in **degress** and heading in **radian**_, which is confusing and it needs to be fixed.

@dsuren1 Thanks. Trying to confirm if those values are radians or degrees. Will revert shorty

**Comment by tonym3nsa** [comment]
> @tonym3nsa A small advice, please make sure to add new commits to the PR whenever you are making changes and don't force push as it makes difficult to track the modifications done

thanks

**Comment by tonym3nsa** [comment]
> @tonym3nsa A small advice, please make sure to add new commits to the PR whenever you are making changes and don't force push as it makes difficult to track the modifications done

thanks

**Comment by dsuren1** [comment]
> @dsuren1 I have reason to believe those figures are in degrees and not radians. For example, look at the below image. The radian value dispatched by the `CHANGE_MAP_VIEW `action (that gets the radian value from cesium) is similar to the values you mentioned are in radians. Or maybe I may be misunderstanding you - please let me know.
> 
> Screenshot 2022-03-09 at 2 11 50 PM png-mh

Hi @tonym3nsa, Yes. The values are in degress. The reason for the confusion is the decimal places.

**Review by dsuren1 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
@tonym3nsa @dsuren1 thank you, tests passed. @tonym3nsa please backport to 2022.01.00.

**Comment by tdipisa** [comment]
@ElenaGallo the backport is to 2022.01.xx. There are no branches with name 2022.01.00

**Pull request merged**